bundles/org.openhab.binding.solarman/src/main/resources/definitions/deye_2mppt.yaml

```diff
@@ -201,7 +202,7 @@ parameters:
       state_class: ""
       uom: ""
       scale: 1
-      rule: 1
+      rule: 5
```

Comment:
Instead of changing the definition files, let's update the code to understand them as they are. As stated elsewhere, these are pulled as is from https://github.com/StephanJoubert/home_assistant_solarman/tree/main/custom_components/solarman/inverter_definitions. Changing them will make it painful to re-sync them back in from upstream.

I assume you could use the presence of `lookup` or/and maybe `isstr`, to determine how to interpret the value.

Reply:
when you do the mapping with lookup, the result is a string. (indicated by rule: 5). rule: 1 means, it is numeric. with rule:1 the definition is wrong and it will not work.
So we have 2 opportunities:
1. leave rule:1 --> only a numeric value can be handled and displayed
2. change to rule:5 the mapping via lookup wokrs an we could display the mapped string value.

Reply:
Looking at the implementation here: https://github.com/StephanJoubert/home_assistant_solarman/blob/d4296b16415dc9187a74fbaf249bfd71417c38aa/custom_components/solarman/parser.py#L98 it seems that the rule determines the type of the value read from Modbus, not the final type of the item. 
In our case we need to determine the type of the item and for this we can look at the type of the values in the lookup and set the item type based on that. 
In all of the cases I've seen (haven't checked them all, though) it's string, so maybe we can simplify the implementation for now and always set it to string if there is a lookup property.

So, let's try to keep the original rule, but either set it to string if there is a lookup, or try to determine the type based on the type of the values in the lookup table.

Reply:
OK, I will find a way to keep the rule:1 in config, but from openhab point of view, we have to create a channel openhab internally with type string, if lookup is present or alternatively if isstr = true.

Reply:
rule:1 kept in config, it works!